### PR TITLE
fix(sdcm/nemesis.py): Provide default values for compaction strategies

### LINKED ISSUE #6153: ModifyTable nemesis - `modify_table_twcs_window_size` generate unrecommended table configuration
```
2023-05-16 02:08:51.356: (DisruptionEvent Severity.NORMAL) period_type=begin event_id=dc481022-67de-4ec6-aee5-e6cfb298acc3: nemesis_name=ModifyTable target_node=Node longevity-lwt-parallel-24h-2023-1-db-node-f7a
072d4-4 [34.249.51.228 | 10.4.2.134] (seed: False)
2023-05-16 02:08:53.036: (InfoEvent Severity.NORMAL) period_type=not-set event_id=a9cb4fdf-cf51-4d5a-af0b-b3950f0a9aee: message=ModifyTablePropertiesCompaction Node longevity-lwt-parallel-24h-2023-1-db-node-f7a0
72d4-4 [34.249.51.228 | 10.4.2.134] (seed: False)

< t:2023-05-16 02:12:13,604 f:nemesis.py      l:1921 c:sdcm.nemesis         p:DEBUG > sdcm.nemesis.ModifyTableMonkey: _modify_table_property: ALTER TABLE cqlstress_lwt_example.blogposts WITH compaction = {'class
': 'TimeWindowCompactionStrategy'};


2023-05-16 02:12:16.562: (DisruptionEvent Severity.ERROR) period_type=end event_id=dc481022-67de-4ec6-aee5-e6cfb298acc3 duration=3m25s: nemesis_name=ModifyTable target_node=Node longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-4 [34.249.51.228 | 10.4.2.134] (seed: False) errors=<Error from server: code=2300 [Query invalid because of configuration issue] message="The setting of default_time_to_live=118079843 and compaction window=86400(s) can lead to 1366 windows, which is larger than the allowed number of windows specified by the twcs_max_window_count (50) parameter. Note that default_time_to_live=0 is also highly discouraged.">
Traceback (most recent call last):
  File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 4370, in wrapper
    result = method(*args[1:], **kwargs)
  File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 2604, in disrupt_modify_table
    disrupt_func()
  File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 2376, in modify_table_compaction
    self._modify_table_property(name="compaction", val=str(prop_val))
  File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 1923, in _modify_table_property
    session.execute(cmd)
  File "/home/ubuntu/scylla-cluster-tests/sdcm/utils/common.py", line 1657, in execute_verbose
    return execute_orig(*args, **kwargs)
  File "cassandra/cluster.py", line 2699, in cassandra.cluster.Session.execute
  File "cassandra/cluster.py", line 5018, in cassandra.cluster.ResponseFuture.result
cassandra.protocol.ConfigurationException: <Error from server: code=2300 [Query invalid because of configuration issue] message="The setting of default_time_to_live=118079843 and compaction window=86400(s) can lead to 1366 windows, which is larger than the allowed number of windows specified by the twcs_max_window_count (50) parameter. Note that default_time_to_live=0 is also highly discouraged.">
```
## Issue description

- [ ] This issue is a regression.
- [ ] It is unknown if this issue is a regression.

*Describe your issue in detail and steps it took to produce it.*

## Impact

*Describe the impact this issue causes to the user.*

## How frequently does it reproduce?

*Describe the frequency with how this issue can be reproduced.*

## Installation details

Kernel Version: 5.15.0-1035-aws
Scylla version (or git commit hash): `2023.1.0~rc5-20230429.a47bcb26e42e` with build-id `d2644a8364f13d14d25be6b9d3c69f84612192bd`

Cluster size: 4 nodes (i3.2xlarge)

Scylla Nodes used in this run:
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-9 (3.250.109.85 | 10.4.1.49) (shards: 7)
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-8 (18.203.223.110 | 10.4.2.52) (shards: 7)
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-7 (34.244.199.69 | 10.4.2.204) (shards: 7)
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-6 (54.154.54.73 | 10.4.2.180) (shards: 7)
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-5 (34.255.6.153 | 10.4.1.2) (shards: 7)
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-4 (34.249.51.228 | 10.4.2.134) (shards: 7)
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-3 (18.202.235.51 | 10.4.2.207) (shards: 7)
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-2 (3.252.206.24 | 10.4.1.104) (shards: 7)
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-16 (52.48.130.233 | 10.4.2.21) (shards: 7)
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-15 (34.253.181.49 | 10.4.2.70) (shards: 7)
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-14 (34.241.112.227 | 10.4.0.168) (shards: 7)
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-13 (63.32.46.61 | 10.4.0.247) (shards: 7)
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-12 (3.250.11.52 | 10.4.3.162) (shards: 7)
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-11 (3.252.163.225 | 10.4.1.200) (shards: 7)
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-10 (3.250.50.18 | 10.4.0.86) (shards: 7)
- longevity-lwt-parallel-24h-2023-1-db-node-f7a072d4-1 (52.212.121.45 | 10.4.0.88) (shards: 7)

OS / Image: `ami-05e7801837cea47d9` (aws: eu-west-1)

Test: `longevity-lwt-parallel-schema-changes-with-disruptive-24h-test`
Test id: `f7a072d4-311b-4f08-ac18-97c9a39b8378`
Test name: `enterprise-2023.1/longevity/longevity-lwt-parallel-schema-changes-with-disruptive-24h-test`
Test config file(s):

- [longevity-lwt-parallel-24h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/1dafb96f428e2691c12e03508dc5024d85552043/test-cases/longevity/longevity-lwt-parallel-24h.yaml)

<details>
<summary>
Logs and commands
</summary>

- Restore Monitor Stack command: `$ hydra investigate show-monitor f7a072d4-311b-4f08-ac18-97c9a39b8378`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=f7a072d4-311b-4f08-ac18-97c9a39b8378)
- Show all stored logs command: `$ hydra investigate show-logs f7a072d4-311b-4f08-ac18-97c9a39b8378`

## Logs:
- **db-cluster-f7a072d4.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/f7a072d4-311b-4f08-ac18-97c9a39b8378/20230516_154750/db-cluster-f7a072d4.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/f7a072d4-311b-4f08-ac18-97c9a39b8378/20230516_154750/db-cluster-f7a072d4.tar.gz)
- **sct-runner-events-f7a072d4.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/f7a072d4-311b-4f08-ac18-97c9a39b8378/20230516_154750/sct-runner-events-f7a072d4.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/f7a072d4-311b-4f08-ac18-97c9a39b8378/20230516_154750/sct-runner-events-f7a072d4.tar.gz)
- **sct-f7a072d4.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/f7a072d4-311b-4f08-ac18-97c9a39b8378/20230516_154750/sct-f7a072d4.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/f7a072d4-311b-4f08-ac18-97c9a39b8378/20230516_154750/sct-f7a072d4.log.tar.gz)
- **monitor-set-f7a072d4.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/f7a072d4-311b-4f08-ac18-97c9a39b8378/20230516_154750/monitor-set-f7a072d4.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/f7a072d4-311b-4f08-ac18-97c9a39b8378/20230516_154750/monitor-set-f7a072d4.tar.gz)
- **loader-set-f7a072d4.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/f7a072d4-311b-4f08-ac18-97c9a39b8378/20230516_154750/loader-set-f7a072d4.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/f7a072d4-311b-4f08-ac18-97c9a39b8378/20230516_154750/loader-set-f7a072d4.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/enterprise-2023.1/job/longevity/job/longevity-lwt-parallel-schema-changes-with-disruptive-24h-test/4/)
</details>

### PR DESCRIPTION
**Pull request by k0machi**
This change fixes an issue where switching a compaction strategy on an
already existing table would result in the new compaction strategy
attempting to use old compaction values, which could be completely
incompatible with said compaction strategy. Now the strategies contain
default values for each per scylla docs.

Fixes #6153

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by k0machi** [comment]
Verification job: [link](https://argus.scylladb.com/test/d4b581da-c465-4c72-94da-9e663cc48b27/runs?additionalRuns[]=0b92c2db-7fd1-49d6-a7fb-e2ef4cd63076)

**Comment by k0machi** [comment]
It seems this is caused by interaction with this nemesis:
```py
    def modify_table_default_time_to_live(self):  # pylint: disable=invalid-name
        """
            The value of this property is a number of seconds. If it is set, Cassandra applies a
            default TTL marker to each column in the table, set to this value. When the table TTL
            is exceeded, Cassandra tombstones the table.
            default: default_time_to_live = 0
        """
        # Select table without columns with "counter" type for this nemesis - issue #1037:
        #    Modify_table nemesis chooses first non-system table, and modify default_time_to_live of it.
        #    But table with counters doesn't support this
        self._modify_table_property(name="default_time_to_live", val=random.randint(864000, 630720000),
```

Would it be a good idea to reset this to 0 after the modification has been done? Seems like this could delete data written after this change.

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -2471,9 +2471,29 @@ def modify_table_compaction(self):
             default: compaction = {'class': 'SizeTieredCompactionStrategy'}
         """
         # TODO: Sub-properties for each of compaction strategies should also be tested
```
[the remark this answers is not in the document]
Comment (k0machi) [reply sdcm/nemesis.py]: Done

**Comment by k0machi** [comment]
I've clamped `modify_table_default_time_to_live` to 49 days because of this, let's see if the nemesis will fail now.

**Comment by k0machi** [comment]
> I've clamped `modify_table_default_time_to_live` to 49 days because of this, let's see if the nemesis will fail now.

Looks good so far, another thing I could do is retrieve this parameter and divide it by 49 to get amount of seconds for the compaction window

**Comment by temichus** [comment]
@fruch , @soyacz , Could you please merge this PR?

**Pull request merged**